### PR TITLE
briar-desktop: init at 0.2.1-beta

### PR DESCRIPTION
**Pull request by onny**
###### Motivation for this change

Packaging [briar](https://briarproject.org/), a decentralized p2p messenger

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/briar-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    mkdir $out
+    cp -a * $out
+    mv $out/bin/eclair-node.sh $out/bin/eclair-node
+    rm $out/bin/eclair-node.bat
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A scala implementation of the Lightning Network";
```
Comment (grwlf) [inline pkgs/applications/networking/instant-messengers/briar-desktop/default.nix:31]: Lightning Network? Isn't it Briar?

**Comment by edrex** [comment]
TODO: @edrex review this

**Comment by prusnak** [comment]
I really doubt the `installPhase` is correct.

**Comment by onny** [comment]
> Not sure if you were thinking to package the upstream jar or build with gradle. For the latter, #17342

Thanks for pinging this PR. I updated it to the latest version and want to package the jar file. Right now I'm looking into how to use makeWrapper for it correctly.

**Comment by onny** [comment]
It's now possible to execute and run briar. Unfortunately it is shipping it's own `tor`-binary so we need to patch it somehow

**Comment by onny** [comment]
Got it patched :) It's working now

**Comment by sebkur** [comment]
@onny I'm a briar-desktop dev and just had a chat with a user trying to run on the briar-desktop jar on NixOS instead of installing your package here. I see you've added this workaround for this:

> It's now possible to execute and run briar. Unfortunately it is shipping it's own tor-binary so we need to patch it somehow

Could you maybe explain or point me to a resource explaining why we cannot run our own tor binary on NixOS? Thanks!

**Comment by chayleaf** [comment]
@sebkur

> Could you maybe explain or point me to a resource explaining why we cannot run our own tor binary on NixOS? Thanks!

1. NixOS isn't FHS-compliant. It doesn't have `/usr/lib`, or `/lib`. This means any binaries compiled for traditional Linux distros won't work out of the box - they have to either be patched, or run in a sandbox.
2. Any self-respecting Linux distro maintainer will try to remove vendored deps *as much as it is feasible*, because vendoring introduces duplication of storage space, duplication of efforts (if version X of library Y is broken/requires a workaround or a security fix, you now have to fix not only the package Y, but also all packages that vendor it). In fact, excessive vendoring is grounds for not packaging something at all (for example, I wanted to package frida, and [I did](https://github.com/chayleaf/frida.nix/), kinda - and due to all the vendoring I'm not going to maintain it, or put it into nixpkgs; plus the vendoring made the codebase so fickle the package didn't even fully work)

**Pull request merged**